### ISSUE TITLE
Login blocked because 2FA isn't enabled, but 2FA shows as enabled in account settings

### ISSUE THREAD

**Issue by LeoMcA** [mozilla/fxa-auth-server#2935]
Note: I can't reproduce this since @gdestuynder turned off 2FA verification during FxA authentication, but I imagine the bug still exists

Steps:
1. Open voice.mozilla.org
2. Click "Log In / Sign Up"
3. Select "Log in with Firefox"
4. Click "Sign In" on Firefox Accounts page (I'm already logged in to Firefox Accounts)
5. Red banner saying 2FA needs to be enabled is displayed
6. Open https://accounts.firefox.com/settings/two_step_authentication
7. Two-step authentication says "Enabled"

@gdestuynder you might be able to fill more information in here as I'm not sure what's going on behind the scenes.

**Comment by gdestuynder** [comment]
note what I turned off is the 2FA enforcement request (`/authorization?acr...`) on the Mozilla IAM relier to FxA

**Comment by hmitsch** [comment]
@LeoMcA thanks for reporting. I sometimes here volunteers reporting this issue and have not gotten a proper handle of it yet.

-Henrik

**Comment by rfk** [comment]
> note what I turned off is the 2FA enforcement request (`/authorization?acr...`) on the Mozilla IAM relier to FxA

Thanks for the report! To clarify the technical details, does this mean basically:

* IAM was directing the user to FxA with `/authorization?acr_values=AAL2`
* FxA prompts the user to sign in, and then rejects it because they don't have 2FA
* But they actually *do* have 2FA enabled, so FxA is behaving weirdly

I expect the key here is this part:

> Click "Sign In" on Firefox Accounts page (I'm already logged in to Firefox Accounts)

Meaning that the user is using an existing login session created *before* 2FA was enabled on their account (and hence not flagged as having been verified with 2FA).  Related issue: https://github.com/mozilla/fxa-auth-server/issues/2922

**Comment by gdestuynder** [comment]
@rfk yes, that sounds exactly like the issue to me. @LeoMcA  does this match your flow?

**Comment by LeoMcA** [comment]
> @LeoMcA does this match your flow?

Yep

> Meaning that the user is using an existing login session created before 2FA was enabled on their account (and hence not flagged as having been verified with 2FA). Related issue: #2922

I'm not sure if that's the case with me - I enabled 2FA almost as soon as the feature was released, IIRC, but I suppose a session could have persisted that long.

**Comment by shane-tomlinson** [comment]
> but I suppose a session could have persisted that long.

FxA sessions last and last and last...

**Comment by shane-tomlinson** [comment]
Paraphrasing an email with @LeoMcA: The the session on the device on which the problem manifests was created 2 months before setting up 2FA. 

The message displayed is indicative of a call to the `/authorization` endpoint on the OAuth server failing because the [session is not 2FA verified](https://github.com/mozilla/fxa-auth-server/blob/40b65eeb05a11437ae32cbf4b2ed06e30d71f261/fxa-oauth-server/lib/routes/authorization.js#L236). Unless we #2922, the correct behavior for this case is to ask users to enter their TOTP code to become 2FA verified.

I'm moving this to the content server for now. I'd like to have #2922, though the current behavior was a policy decision and changing that policy that will need to be discussed with the @jvehent and the OpSec team.
